### PR TITLE
Add drag completion callback to Draggable.

### LINKED ISSUE #10350: onDraggableCompleted callback for Draggable
There currently does not seem to be a way to know when a `Draggable` has been successfully dragged to a `DragTarget` from the point of view of the original owner of the draggable. They'll know if the drag failed with `onDraggableCanceled`, but no notification is sent in case of success. This would be very useful to clean up any leftovers from the now-gone draggable.

/cc @Hixie

### PR DESCRIPTION
**Pull request by pylaligand**
Fixes #10350

### CODE REVIEW

**Comment by pylaligand** [comment]
I need to update some automated and manual tests, but throwing it out there to get some comments first.

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 1]
```diff
@@ -94,7 +94,8 @@ class Draggable<T> extends StatefulWidget {
     this.affinity,
     this.maxSimultaneousDrags,
     this.onDragStarted,
```
Comment (pylaligand) [inline packages/flutter/lib/src/widgets/drag_target.dart]: The naming of the callbacks looks inconsistent.

**Comment by Hixie** [comment]
This will need tests.

I agree that the naming is inconsistent. I would call this new one onDragCompleted. We could try renaming the other one but to do so we'd have to follow the advice here: https://flutter.io/design-principles/#handling-breaking-changes

**Comment by Hixie** [comment]
(General approach seems fine.)

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 2]
```diff
@@ -182,6 +183,9 @@ class Draggable<T> extends StatefulWidget {
   /// callback is still in the tree.
   final DraggableCanceledCallback onDraggableCanceled;
 
+   /// Called when the draggable is dropped and accepted by a [DragTarget].
+  final VoidCallback onDraggableCompleted;
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/drag_target.dart]: these docs should be more detailed, e.g. it should cover information about the lifetime of the widget that provided this callback relative to when the callback is fired, etc.

Comment (pylaligand) [inline packages/flutter/lib/src/widgets/drag_target.dart]: Done. I copied most of what was in onDraggableCanceled as it seemed applicable.

**Comment by pylaligand** [comment]
Tests added.

**Comment by pylaligand** [comment]
LG?

**Comment by Hixie** [comment]
LGTM

**Pull request merged**